lib/CodeGen/PagerandoBinning.cpp

```diff
@@ -8,14 +8,16 @@
 //===----------------------------------------------------------------------===//
 //
 // This pass assigns pagerando-enabled functions to bins. Normal functions
-// (and pagerando wrappers) are put into to the default bin #0.
+// (and pagerando wrappers) are not assigned to a bin.
 // Function sizes are estimated by adding up the size of all instructions
 // inside the corresponding MachineFunction. The default bin size is 4KB.
 // The current bin allocation strategy is a greedy algorithm that, for every
 // function, picks the bin with the smallest remaining free space that still
 // accommodates the function. If such a bin does not exist, a new one is
 // created. Functions that are larger than the default bin size are assigned to
 // a new bin which forces the expansion of said bin.
+// Note: Because this pass estimates function sizes it should run as late as
+// possible.
```

Comment:
You mean the optimizer right? This is binning.
I think for the optimizer it's explicitly stated in the code. Maybe even enforced?
```cpp
MachineFunctionProperties getRequiredProperties() const override {
    return MachineFunctionProperties().set(
        MachineFunctionProperties::Property::TracksLiveness);
  }
```